### PR TITLE
KRP-1232 Compliance questions

### PR DESCRIPTION
**Pull request by p-janik**
https://linear.app/ageras-group/issue/KRP-1232/%5Bmock%5D-business-identification-add-compliance-endpoints

### AC:

add an endpoint for [listing compliance questions](https://docs.solarisgroup.com/api-reference/identity/identifications/#tag/Business-identifications/paths/~1v1~1businesses~1{business_id}~1identifications~1{business_identification_id}~1legal_identification~1questions/get) 

add an endpoint for [creating an answer](https://docs.solarisgroup.com/api-reference/identity/identifications/#tag/Business-identifications/paths/~1v1~1businesses~1{business_id}~1identifications~1{business_identification_id}~1legal_identification~1questions~1{question_id}~1answers/post) to one question

update answered question entity

[mark legal identification](https://docs.solarisgroup.com/api-reference/identity/identifications/#tag/Business-identifications/paths/~1v1~1businesses~1{business_id}~1identifications~1{business_identification_id}~1legal_identification~1mark_as_ready/patch) as ready

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -904,6 +904,29 @@ router.get(
   safeRequestHandler(businessesAPI.retrieveBusinessIdentification)
 );
 
+// COMPLIANCE QUESTIONS
+
+router.post(
```
Comment (lera) [inline src/app.ts]: get

File: src/routes/business/complianceQuestions.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+      ],
+    });
+    return;
+  }
+
+  question.answer_id = generateID();
+  question.answer_text = answerText;
+  question.answered_at = new Date().toISOString();
+
+  await saveBusiness(business);
```
Comment (lera) [inline src/routes/business/complianceQuestions.ts:84]: you do not save question in BI

File: src/routes/business/complianceQuestions.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+  res.status(200).send(question);
```
Comment (lera) [inline src/routes/business/complianceQuestions.ts]: 201

File: src/routes/business/complianceQuestions.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    });
+    return;
+  }
+
+  businessIdentification.legal_identification_missing_information_details =
+    null;
+  businessIdentification.legal_identification_missing_information = [];
+  businessIdentification.legal_identification_status =
+    LegalIdentificationStatus.PENDING;
+  businessIdentification.status = BusinessIdentificationStatus.PENDING;
```
Comment (lera) [inline src/routes/business/complianceQuestions.ts]: this status remained 'created' on mocksolaris 🤷‍♀️

File: src/app.ts (hunk 1 above)
Comment (p-janik) [reply src/app.ts]:
```suggestion
router.get(
```

File: src/routes/business/complianceQuestions.ts (hunk 2 above)
Comment (p-janik) [reply src/routes/business/complianceQuestions.ts:84]: I do, via reference

File: src/routes/business/complianceQuestions.ts (hunk 3 above)
Comment (p-janik) [inline src/routes/business/complianceQuestions.ts]:
```suggestion
  res.status(201).send(question);
```

File: src/helpers/types.ts [hunk 5]
```diff
@@ -346,8 +347,23 @@ export type BusinessIdentification = {
   legal_identification_missing_information_details?: string;
   legal_representatives: LegalRepresentativeIdentificationResponse[];
   legal_identification_missing_information: string[];
+  meta?: {
```
Comment (lera) [inline src/helpers/types.ts:350]: it shouldn't be included into BI response on 'retrieve BI' request

File: src/routes/business/complianceQuestions.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    });
+    return;
+  }
+
+  businessIdentification.legal_identification_missing_information_details =
+    null;
+  businessIdentification.legal_identification_missing_information = [];
+  businessIdentification.legal_identification_status =
+    LegalIdentificationStatus.PENDING;
+  businessIdentification.status = BusinessIdentificationStatus.CREATED;
```
Comment (lera) [inline src/routes/business/complianceQuestions.ts]: you do not need to update `businessIdentification.status`

**Review by lera — APPROVED** [review]

**Pull request merged**